src/gui/qgstemporalcontrollerwidget.cpp

```diff
@@ -167,6 +178,58 @@ void QgsTemporalControllerWidget::setWidgetStateFromProject()
   mNavigationObject->setTemporalRangeCumulative( QgsProject::instance()->timeSettings()->isTemporalRangeCumulative() );
 }
 
+void QgsTemporalControllerWidget::mNavigationOff_clicked()
+{
+  QgsProject::instance()->writeEntry( QStringLiteral( "TemporalControllerWidget" ), QStringLiteral( "/NavigationMode" ),
+                                      static_cast<int>( QgsTemporalNavigationObject::NavigationOff ) );
+
+  mNavigationObject->setNavigationMode( QgsTemporalNavigationObject::NavigationOff );
+  setWidgetStateFromNavigationMode( QgsTemporalNavigationObject::NavigationOff );
+}
+
+void QgsTemporalControllerWidget::mNavigationFixedRange_clicked()
+{
+  QgsProject::instance()->writeEntry( QStringLiteral( "TemporalControllerWidget" ), QStringLiteral( "/NavigationMode" ),
+                                      static_cast<int>( QgsTemporalNavigationObject::FixedRange ) );
+
+  mNavigationObject->setNavigationMode( QgsTemporalNavigationObject::FixedRange );
+  setWidgetStateFromNavigationMode( QgsTemporalNavigationObject::FixedRange );
+}
+
+void QgsTemporalControllerWidget::mNavigationAnimated_clicked()
+{
+  QgsProject::instance()->writeEntry( QStringLiteral( "TemporalControllerWidget" ), QStringLiteral( "/NavigationMode" ),
+                                      static_cast<int>( QgsTemporalNavigationObject::Animated ) );
+
+  mNavigationObject->setNavigationMode( QgsTemporalNavigationObject::Animated );
+  setWidgetStateFromNavigationMode( QgsTemporalNavigationObject::Animated );
+}
+
+void QgsTemporalControllerWidget::setWidgetStateFromNavigationMode( const QgsTemporalNavigationObject::NavigationMode mode )
+{
+  mNavigationOff->setChecked( mode == QgsTemporalNavigationObject::NavigationOff );
```

Comment:
This would be nicer if you used a stacked widget -- then you'd just set the current index of the stacked widget here. It would help avoid the extra padding which shows up when you hide elements like this (Because qt still leaves the layout padding for them)

Reply:
True -- I thought it was better to go down that path to avoid duplicating the QDateTimeEdit widgets. 

If we opt-in stacked widgets, we'll have to maintain a pair of QDateTimeEdit widgets for the start datetime and end datetime.

Reply:
I think duplication is fine, and is outweighed by the benefits of the stacked widget approach

Reply:
Also, if we go stacked it'll be easier to add the double ranged slider in future ;)